tm2/pkg/crypto/keys/client/add.go

```diff
@@ -204,6 +198,29 @@ func execAdd(cfg *AddCfg, args []string, io commands.IO) error {
 	return nil
 }
 
+// Prompts for a password, with confirmation.
+func getPassphraseWithConfirmation(io commands.IO, insecurePasswordStdin bool) (string, error) {
+	pw, err := io.GetPassword("Enter a passphrase to encrypt your private key to disk:", insecurePasswordStdin)
+	if err != nil {
+		return "", fmt.Errorf("unable to get provided passphrase, %w", err)
+	}
+
+	prompt2 := "Repeat the passphrase:"
+	if pw == "" {
+		prompt2 = "WARNING: a key with no passphrase will be stored unencrypted.\n" +
+			"This unsafe for any key used on-chain.\n" + prompt2
+	}
+	pw2, err := io.GetPassword(prompt2, insecurePasswordStdin)
+	if err != nil {
+		return "", fmt.Errorf("unable to get provided passphrase, %w", err)
+	}
+	if pw != pw2 {
```

Comment:
Maybe I'm exaggerating, but this is one of the few instances I would use subtle, to avoid timing comparisons. Nitpick more than anything